pkg/operation/botanist/addons/nginxingress/nginxingress.go

```diff
@@ -0,0 +1,699 @@
+			},
+			Spec: corev1.ServiceSpec{
+				Type:                     corev1.ServiceTypeLoadBalancer,
+				LoadBalancerSourceRanges: n.values.LoadBalancerSourceRanges,
+				ExternalTrafficPolicy:    n.values.ExternalTrafficPolicy,
+				Ports: []corev1.ServicePort{
+					{
+						Name:       "http",
+						Port:       servicePortControllerHttp,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromInt(int(containerPortControllerHttp)),
+					},
+					{
+						Name:       "https",
+						Port:       servicePortControllerHttps,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromInt(int(containerPortControllerHttps)),
+					},
+				},
+				Selector: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyRelease:           labelValueAddons,
+					labelKeyComponent:         labelValueController,
+				},
+			},
+		}
+
+		serviceBackend = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: serviceNameBackend,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyComponent:         labelValueBackend,
+					labelKeyRelease:           labelValueAddons,
+				},
+				Namespace: n.namespace,
+			},
+			Spec: corev1.ServiceSpec{
+				Type: corev1.ServiceTypeClusterIP,
+				Ports: []corev1.ServicePort{{
+					Port:       servicePortBackend,
+					TargetPort: intstr.FromInt(int(containerPortBackend)),
+				}},
+				Selector: map[string]string{
+					v1beta1constants.LabelApp: labelAppValue,
+					labelKeyComponent:         labelValueBackend,
+					labelKeyRelease:           labelValueAddons,
+				},
+			},
+		}
+
+		ingressClass   *networkingv1.IngressClass
+		vpa            *vpaautoscalingv1.VerticalPodAutoscaler
+		roleBindingPSP *rbacv1.RoleBinding
+	)
+
+	if n.values.VPAEnabled {
+		updateMode := vpaautoscalingv1.UpdateModeAuto
+		vpa = &vpaautoscalingv1.VerticalPodAutoscaler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      vpaName,
+				Namespace: n.namespace,
+			},
+			Spec: vpaautoscalingv1.VerticalPodAutoscalerSpec{
+				TargetRef: &autoscalingv1.CrossVersionObjectReference{
+					APIVersion: appsv1.SchemeGroupVersion.String(),
+					Kind:       "Deployment",
+					Name:       deploymentController.Name,
+				},
+				UpdatePolicy: &vpaautoscalingv1.PodUpdatePolicy{
+					UpdateMode: &updateMode,
+				},
+				ResourcePolicy: &vpaautoscalingv1.PodResourcePolicy{
+					ContainerPolicies: []vpaautoscalingv1.ContainerResourcePolicy{
+						{
+							ContainerName: vpaautoscalingv1.DefaultContainerResourcePolicy,
+							MinAllowed: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("100m"),
+								corev1.ResourceMemory: resource.MustParse("128Mi"),
+							},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	if !n.values.PSPDisabled {
+		roleBindingPSP = &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      roleBindingPSPName,
+				Namespace: n.namespace,
```

Comment:
```suggestion
				Namespace: addonNamespace,
```
nginx-ingress-addon should be deployed in `kube-system` namespace of the shoot.